### PR TITLE
Clean up some aggregation tests

### PR DESCRIPTION
**Pull request by nik9000**
This rewrites two tests for aggregations to use `AggregatorTestCase`'s
simpler way of making `Aggregator`s, allowing us to remove a ctor on
`ProductionAggregationContext` that we weren't happy about. Now there is
only a single test call to `ProductionAggregationContext` and we can
remove that soon.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/search/aggregations/AggregationCollectorTests.java [hunk 1]
```diff
@@ -19,54 +19,109 @@
...
+import org.elasticsearch.search.aggregations.metrics.TopHitsAggregationBuilder;
 
 import java.io.IOException;
 
+import static org.mockito.Matchers.any;
 import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
 
-public class AggregationCollectorTests extends ESSingleNodeTestCase {
+public class AggregationCollectorTests extends AggregatorTestCase {
```
Comment (not-napoleon) [inline server/src/test/java/org/elasticsearch/search/aggregations/AggregationCollectorTests.java:40]: Oh, hey, moving this off of `ESSingleNodeTestCase` is a win!

File: server/src/test/java/org/elasticsearch/search/aggregations/AggregationCollectorTests.java [hunk 2]
```diff
@@ -19,54 +19,109 @@
...
+
+    public void testScriptNeedsScoresIfScriptNeedsScores() throws IOException {
+        assertTrue(needsScores(termsBuilder().script(new Script("with_scores"))));
+    }
 
-    public void testNeedsScores() throws Exception {
-        IndexService index = createIndex("idx");
-        client().prepareIndex("idx").setId("1").setSource("f", 5).execute().get();
-        client().admin().indices().prepareRefresh("idx").get();
+    public void testSubScriptDoesntNeedScoresIfSubScriptDoesntNeedScores() throws IOException {
```
Comment (not-napoleon) [inline server/src/test/java/org/elasticsearch/search/aggregations/AggregationCollectorTests.java:57]: I am not a fan of this method of documenting tests. CamelCase sentences are hard to read, and with many similar names it's easy to miss the differences between two of them.  Maybe just name this something like `testSubScriptNoScore` and add a javadoc noting what it's intended to test?  Same for the other very long test names.

**Review by not-napoleon — APPROVED** [review]
LGTM.  Left a nit about test naming, but it's a matter of taste.

File: server/src/test/java/org/elasticsearch/search/aggregations/AggregationCollectorTests.java (hunk 2 above)
Comment (nik9000) [reply server/src/test/java/org/elasticsearch/search/aggregations/AggregationCollectorTests.java:57]: I like having a test for each thing. I could go either way on the long names. I'll change them.

**Pull request merged**